lib/plugins/logviewer/admin.php

```diff
@@ -150,5 +123,86 @@ protected function getFacilities()
         return $facilities;
     }
 
+    /**
+     * Read the lines of the logfile and return them as an array
+     *
+     * @param string $logfilePath
+     * @return array
+     */
+    protected function getLogLines($logfilePath)
+    {
+        global $lang;
+
+        $lines = [];
+        $size = filesize($logfilePath);
+        
+        if ($size <= self::MAX_READ_SIZE) {
+            $lines = file($logfilePath);
+        } else {
+            $fp = fopen($logfilePath, 'r');
+            if ($fp !== null) {
+                if (fseek($fp, -self::MAX_READ_SIZE, SEEK_END) == 0) {
+                    if (false !== ($d = fread($fp, self::MAX_READ_SIZE))) {
```

Comment:
try to avoid these deeply nested if statements. Instead check the error condition first and abort early if an error occurs. (I am also not sure we really need to differentiate between open and read errors in this case)